packages/vite/src/node/optimizer/scan.ts

```diff
@@ -239,6 +239,10 @@ function orderedDependencies(deps: Record<string, string>) {
 }
 
 function globEntries(pattern: string | string[], config: ResolvedConfig) {
+  const resolvedPatterns = Array.isArray(pattern) ? pattern : [pattern]
```

Comment:
micro nit: we have the `arraify` util for this.

Reply:
@Jinjiang let's do this change and then we can merge it

Reply:
Done